File: typescript/src/contracts/types/process_start_request.ts
export class ProcessStartRequest<TPayload> {

  public readonly correlationId: string
  public readonly parentProcessInstanceId: string
  public readonly payload: TPayload;

  constructor(correlationId: string, parentProcessInstanceId: string, payload: TPayload) {
    this.correlationId = correlationId;
    this.parentProcessInstanceId = parentProcessInstanceId;
    this.payload = payload;
  }

}

File: dotnet/src/contracts/types/ProcessStartRequest.cs
namespace ProcessEngine.Client.Contracts
{
    public class ProcessStartRequest<TPayload>
        where TPayload: new()
    {
        public ProcessStartRequest() {
            this.Payload = new TPayload();
        }

        public ProcessStartRequest(string correlationId, string parentProcessInstanceId, TPayload payload = default(TPayload))
        {
            this.CorrelationId = correlationId;
            this.ParentProcessInstanceId = parentProcessInstanceId;

            this.Payload = payload == null
                ? new TPayload()
                : payload;
        }

        public string CorrelationId { get; private set; }

        public string ParentProcessInstanceId { get; private set; }

        public TPayload Payload { get; private set; }
    }
}
